app/com/linkedin/drelephant/tuning/hbt/FitnessManagerHBT.java

```diff
@@ -163,12 +172,24 @@ protected void checkToDisableTuning(Set<JobDefinition> jobDefinitionSet) {
       if (reachToNumberOfThresholdIterations(tuningJobExecutionParamSets, jobDefinition)) {
         disableTuning(jobDefinition, "User Specified Iterations reached");
       }
-      if (areHeuristicsPassed(tuningJobExecutionParamSets)) {
-        disableTuning(jobDefinition, "All Heuristics Passed1");
+      int autoAppliedExecution = 0;
+      for (TuningJobExecutionParamSet tuningJobExecutionParam : tuningJobExecutionParamSets) {
+        if (tuningJobExecutionParam.jobSuggestedParamSet.isParamSetSuggested
+            && !tuningJobExecutionParam.jobSuggestedParamSet.paramSetState.equals(
+            JobSuggestedParamSet.ParamSetStatus.DISCARDED)) {
+          autoAppliedExecution++;
+        }
+      }
+      if (isDebugEnabled) {
+        logger.debug(" Total number of executions after auto applied enabled " + autoAppliedExecution);
+      }
+      //Minimum three execution needed for HBT to do some resource optimization
+      if (areHeuristicsPassed(tuningJobExecutionParamSets) && autoAppliedExecution >= MINIMUM_HBT_EXECUTION) {
+        disableTuning(jobDefinition, "All Heuristics Passed");
       }
     }
-    Long currentTimeAfter= System.currentTimeMillis();
-    logger.info(" Total time taken by disable tuning " + (currentTimeAfter-currentTimeBefore));
+    Long currentTimeAfter = System.currentTimeMillis();
```

Comment:
nanoTime should be preferred over currentTimeMillis when calculating elapsed time. Can be taken care of in the next PR though.

Reply:
Can be taken care of in the next PR